src/world/partner/bombette.c

```diff
@@ -1,33 +1,50 @@
 #include "common.h"
+#include "../src/world/partners.h"
 
 extern s32 D_802BE934;
-extern f64 D_802BE900_319650;
+//extern f64 D_802BE900_319650;
 
-extern struct unkPartnerStruct* D_802BE89C_3195EC;
+//extern struct unkPartnerStruct* D_802BE89C_3195EC;
+extern s32 D_802BE92C;
+extern s32 D_802BE928;
+extern s32 D_802BE924;
+extern s32 D_802BE920;
+extern unkPartnerStruct* D_802BE89C_3195EC;
+
+ApiStatus func_802BD338_318088(Evt* evt, s32 arg1);
+ApiStatus func_802BD758_3184A8(Evt* evt, s32 isInitialCall);
+ApiStatus func_802BE4E8_319238(Evt* script, s32 isInitialCall);
+ApiStatus func_802BD300_318050(Evt* script, s32 isInitialCall);
 
 INCLUDE_ASM(s32, "world/partner/bombette", func_802BD100_317E50);
 
-INCLUDE_ASM(s32, "world/partner/bombette", func_802BD2D8_318028);
+void func_802BD2D8_318028(Npc* npc) {
+    npc->collisionHeight = 28;
+    npc->collisionRadius = 24;
+    D_802BE928 = 0;
+    D_802BE924 = 0;
+    
+}
 
 ApiStatus func_802BD300_318050(Evt* script, s32 isInitialCall) {
-    Npc* unk = script->owner2.npc; // todo what is this
+    Npc* npc = script->owner2.npc;
 
     if (isInitialCall) {
-        partner_init_get_out(unk);
+        partner_init_get_out(npc);
     }
-    return partner_get_out(unk) != 0;
+    return partner_get_out(npc) != 0;
 }
 
-s32 func_802BD338_318088(Evt* evt, s32 arg1) {
+ApiStatus func_802BD338_318088(Evt* script, s32 isInitialCall) {
     PlayerData* playerData = &gPlayerData;
     Npc* npc;
     f32 sp10;
     f32 sp14;
     f32 tempY;
     Entity* entity;
 
-    npc = evt->owner2.npc;
-    if (arg1 != 0) {
+    npc = script->owner2.npc;
+    if (isInitialCall != 0) {
```

Comment:
I know you didn't write this function in this PR, but could you remove the != 0 here? And put Npc* npc = ...